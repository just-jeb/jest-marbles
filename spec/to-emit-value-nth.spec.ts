import { cold, Scheduler } from '../index';

describe('toEmitValueNth matcher test', () => {
  describe('positive cases', () => {
    it('Should pass if observable completes', () => {
      const a$ = cold('a-b-|');

      expect(a$).toEmitValueNth('b', 1);
    });

    it('Should pass if observable never completes', () => {
      const a$ = cold('a-b-');

      expect(a$).toEmitValueNth('b', 1);
    });
  });

  describe('negative cases', () => {
    it('Should fail if other value emits', () => {
      const a$ = cold('a-b-|');

      expect(a$).toEmitValueNth('a', 1);
    });

    it('Should fail if no values emitted', () => {
      const a$ = cold('---');

      expect(a$).toEmitValueNth('a', 1);
    });

    it('Should fail if no values emitted and completed', () => {
      const a$ = cold('---|');

      expect(a$).toEmitValueNth('a', 1);
    });

    afterEach(() => {
      expect(() => Scheduler.flush()).toThrowErrorMatchingSnapshot();
      Scheduler.init();
    });
  });
});
