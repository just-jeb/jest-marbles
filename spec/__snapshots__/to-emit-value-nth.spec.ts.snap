// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toEmitValueNth matcher test negative cases Should fail if no values emitted 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toBeNotifications([22m[32mexpected[39m[2m)[22m

Expected notifications to be:
  [32m"(a|)"[39m
But got:
  [31m""[39m

Difference:

[32m- Expected[39m
[31m+ Received[39m

[32m- (a|)[39m"
`;

exports[`toEmitValueNth matcher test negative cases Should fail if no values emitted and completed 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toBeNotifications([22m[32mexpected[39m[2m)[22m

Expected notifications to be:
  [32m"---(a|)"[39m
But got:
  [31m"---|"[39m

Difference:

[32m- Expected[39m
[31m+ Received[39m

[32m- ---(a|)[39m
[31m+ ---|[39m"
`;

exports[`toEmitValueNth matcher test negative cases Should fail if other value emits 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toBeNotifications([22m[32mexpected[39m[2m)[22m

Expected notifications to be:
  [32m"--(a|)"[39m
But got:
  [31m"--(b|)"[39m

Difference:

[32m- Expected[39m
[31m+ Received[39m

[32m- --(a|)[39m
[31m+ --(b|)[39m"
`;
